x-pack/plugins/security_solution_serverless/server/endpoint/services/metering_service.ts

```diff
@@ -7,29 +7,49 @@
 
 import type { AggregationsAggregate, SearchResponse } from '@elastic/elasticsearch/lib/api/types';
 import type { ElasticsearchClient } from '@kbn/core/server';
-// import { ENDPOINT_HEARTBEAT_INDEX } from '@kbn/security-solution-plugin/common/endpoint/constants';
+import { ENDPOINT_HEARTBEAT_INDEX } from '@kbn/security-solution-plugin/common/endpoint/constants';
 import type { EndpointHeartbeat } from '@kbn/security-solution-plugin/common/endpoint/types';
 
+import { ProductLine, type ProductTier } from '../../../common/product';
+
 import type { UsageRecord, MeteringCallbackInput } from '../../types';
+import type { ServerlessSecurityConfig } from '../../config';
 
 // 1 hour
 const SAMPLE_PERIOD_SECONDS = 3600;
-// const THRESHOLD_MINUTES = 30;
+const THRESHOLD_MINUTES = 30;
 
 export class EndpointMeteringService {
-  public async getUsageRecords({
+  private tier: ProductTier | undefined;
+
+  public getUsageRecords = async ({
     taskId,
     cloudSetup,
     esClient,
     abortController,
     lastSuccessfulReport,
-  }: MeteringCallbackInput): Promise<UsageRecord[]> {
+    config,
+  }: MeteringCallbackInput): Promise<UsageRecord[]> => {
+    this.setTier(config);
+
     const heartbeatsResponse = await this.getHeartbeatsSince(
       esClient,
       abortController,
       lastSuccessfulReport
     );
 
+    if (!heartbeatsResponse?.hits?.hits) {
+      return [];
+    }
+
+    if (!this.tier) {
```

Comment:
the alternative is to default to `essentials` but this might lead to overbilling so I went with the underbilling option. open to discussion on this.